### PR TITLE
cmd, ethstats, les, mobile, params: native netstats

### PR DESCRIPTION
**Pull request by karalabe**
`geth --ethstats "golang <3:secret@ethstats.net"`

Imgur

### CODE REVIEW

File: core/events.go [hunk 1]
```diff
@@ -38,9 +38,6 @@ type PendingLogsEvent struct {
 // PendingStateEvent is posted pre mining and notifies of pending state changes.
 type PendingStateEvent struct{}
 
-// NewBlockEvent is posted when a block has been imported.
-type NewBlockEvent struct{ Block *types.Block }
-
```
Comment (fjl) [inline core/events.go]: Why is this change included?

Comment (karalabe) [inline core/events.go]: Because I discovered through this PR that we don't use that event. It's dead code. So I didn't want to open an entire PR just to delete 2 lines of dead code.

Comment (fjl) [inline core/events.go]: You wouldn't believe how much dead code we have. IMHO adding a netstats client should not change package core.

Comment (karalabe) [inline core/events.go]: IMHO a contribution that cleans junk should always be welcome :D :P

Comment (fjl) [inline core/events.go]: OK, but please make it a separate commit. Same for the miner changes.

Comment (karalabe) [inline core/events.go]: The miner is needed by this PR, so that is in-context.

**Comment by karalabe** [comment]
@fjl Both `core` and `miner` changes removed from this PR. PTAL

**Comment by karalabe** [comment]
I originally started out by using the API, but hit a lot of missing things, some which were already added by you in your pending PR, some not. Since there was no easy way to get your changes in and extend them too (especially when it merge conflicts and you're also working on it), I did it with the old API (should still be compatible with your PR). When your PR is in, I'll gladly submit a followup to transition this to the Go APIs.

File: ethstats/ethstats.go [hunk 2]
```diff
@@ -0,0 +1,462 @@
...
+
+// New returns a monitoring service ready for stats reporting.
+func New(url string, ethServ *eth.Ethereum, lesServ *les.LightEthereum) (*Service, error) {
+	// Parse the netstats connection url
+	re := regexp.MustCompile("([^:@]*)(:([^@]*))?@(.+)")
+	parts := re.FindStringSubmatch(url)
+	if len(parts) != 5 {
+		return nil, fmt.Errorf("invalid netstats url: \"%s\", should be nodename:secret@host:port", url)
+	}
+	// Allow passwordless login to public ethstats servers
```
Comment (SjonHortensius) [inline ethstats/ethstats.go]: What's the use of these passwords when they are hardcoded? Why not contact both websites and have them remove the password requirement instead?

Comment (karalabe) [inline ethstats/ethstats.go]: These are being debated for now. I would personally just remove passwords from public listings. They were enacted as a sort of deterrent to avoid too many people listing themselves. All in all I don't think that would be the case tbh. That was relevant when Ethereum was born and there were no tools/explorers/whatnot, but now with Mist and other UIs, I think only a few people would really want to list.

Comment (SjonHortensius) [inline ethstats/ethstats.go]: Right, that's what I thought. The _deterring people to list_ reason seems to directly contradict this PR which seeks to **increase** the number of people listing. I like this PR btw; I wanted to list my node as well; but installing the stats client was too much of a hassle.
If the password requirement of both sites could be removed that would be awesome.

Comment (karalabe) [inline ethstats/ethstats.go]: Passwords have been removed n by popular request.
